sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/AppendBlobItem.java

```diff
@@ -37,6 +37,7 @@ public AppendBlobItem(AppendBlobAppendBlockHeaders generatedHeaders) {
     public AppendBlobItem(AppendBlobAppendBlockFromUrlHeaders generatedHeaders) {
         this.lastModified = generatedHeaders.lastModified();
         this.contentMD5 = generatedHeaders.contentMD5();
+        this.encryptionKeySha256 = generatedHeaders.encryptionKeySha256();
         this.blobAppendOffset = generatedHeaders.blobAppendOffset();
         this.blobCommittedBlockCount = generatedHeaders.blobCommittedBlockCount();
     }
```

Comment:
You could consolidate these methods into one by taking a HttpHeaders object. Here's what I did for RestParity
    /**
     * Creates a new FileSmbProperties object from HttpHeaders
     * @param httpHeaders The headers to construct FileSmbProperties from
     */
    FileSmbProperties(HttpHeaders httpHeaders) {
        this.filePermissionKey = httpHeaders.value(FileConstants.HeaderConstants.FILE_PERMISSION_KEY);
        this.ntfsFileAttributes = NtfsFileAttributes.toAttributes(httpHeaders.value(FileConstants.HeaderConstants.FILE_ATTRIBUTES));
        this.fileCreationTime = OffsetDateTime.parse(httpHeaders.value(FileConstants.HeaderConstants.FILE_CREATION_TIME));
        this.fileLastWriteTime = OffsetDateTime.parse(httpHeaders.value(FileConstants.HeaderConstants.FILE_LAST_WRITE_TIME));
        this.fileChangeTime = OffsetDateTime.parse(httpHeaders.value(FileConstants.HeaderConstants.FILE_CHANGE_TIME));
        this.fileId = httpHeaders.value(FileConstants.HeaderConstants.FILE_ID);
        this.parentId = httpHeaders.value(FileConstants.HeaderConstants.FILE_PARENT_ID);
    }